### PR TITLE
[29220] Reuse resizer component in wp-resizer

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/29220

The resizer component only allowed the absolute distance between initial position and current, but for wp-resizer, we needed the relative delta between each move

### CODE REVIEW

File: frontend/src/app/components/resizer/wp-resizer.component.ts [hunk 1]
```diff
@@ -26,29 +26,36 @@
...
-import {Component, ElementRef, HostListener, Input, OnInit} from '@angular/core';
-import {distinctUntilChanged} from 'rxjs/operators';
+import {AfterViewInit, Component, ElementRef, Input, OnInit} from '@angular/core';
+import {debounceTime, distinctUntilChanged} from 'rxjs/operators';
 import {TransitionService} from '@uirouter/core';
 import {MainMenuToggleService} from "core-components/main-menu/main-menu-toggle.service";
 import {BrowserDetector} from "core-app/modules/common/browser/browser-detector.service";
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import {ResizeDelta} from "core-app/modules/common/resizer/resizer.component";
+import {fromEvent} from "rxjs";
```
Comment (openprojectci) [inline frontend/src/app/components/resizer/wp-resizer.component.ts:37]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/95e595bf194301ef92e7cb393bfec5375ab09b02/frontend/src/app/components/resizer/wp-resizer.component.ts#L37"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/common/resizer/resizer.component.ts [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
+    x:number;
+    y:number;
+  };
+
+  // Relative difference from last position
+  relative:{
+    x:number;
+    y:number;
+  };
 }
```
Comment (openprojectci) [inline frontend/src/app/modules/common/resizer/resizer.component.ts:19]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/95e595bf194301ef92e7cb393bfec5375ab09b02/frontend/src/app/modules/common/resizer/resizer.component.ts#L19"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Review by HDinger — CHANGES_REQUESTED** [review]
The `npm run serve` currently crashes on start. This is due to this place: https://github.com/opf/openproject/blob/dev/frontend/src/app/components/resizer/main-menu-resizer.component.ts#L93 . 

The WP resizer seems to work fine. I could not test whether it still works on iOS, but I remember that we had a lot os issues there. Did you check that?

**Comment by oliverguenther** [comment]
I have no IOS so cannot test this. I added the `touchmove` events to resizer. Otherwise, I don't see what could have changed, all the behavior is identical otherwise

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/95e595bf194301ef92e7cb393bfec5375ab09b02/frontend/src/app/components/resizer/main-menu-resizer.component.ts#L36">frontend/src/app/components/resizer/main-menu-resizer.component.ts#L36</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by HDinger** [comment]
Yes, I thought so too. If no one can test it, it has to be done during the QA.

**Pull request merged**